ppr-ui/src/utils/mhr-api-helper.ts

```diff
@@ -648,6 +669,20 @@ export function delayActions (milliseconds: number): Promise<any> {
   return new Promise(resolve => setTimeout(resolve, milliseconds))
 }
 
+// Delete empty properties and objects and used to cleanup payload before API submission
+export function deleteEmptyProperties (obj) {
+  for (const key in obj) {
+    if (typeof obj[key] === 'object') {
+      deleteEmptyProperties(obj[key]) // recursively process nested objects
+      if (Object.keys(obj[key]).length === 0) {
+        delete obj[key] // delete empty nested object
+      }
+    } else if (obj[key] === null || obj[key] === undefined || obj[key] === '') {
+      delete obj[key] // delete empty property value
+    }
+  }
+}
+
```

Comment:
![image](https://github.com/bcgov/ppr/assets/77707952/bcb33021-036d-48ec-a96c-b28f75110b4c)

I am having an issue with this function here if you can please take a look.

I went to file a notice of caution entered in most of the info, pressed on "Date in time in the past" but didn't enter anything and pressed submit.

Reply:
Probably just needs an optional chaining on a value it's trying to validate.

Reply:
Thanks for pre-QA 😄 looking into it.

Reply:
Fixed the validation for EffectiveDateComponent. Thanks for the catch!